src/components/Input/index.jsx

```diff
@@ -0,0 +1,107 @@
+import { useEffect, useRef, useState } from 'react';
+
+import validate from './validate';
+
+export default function Input({ type = null, state = '', ...props }) {
+  // For controlling input
+  const [inputValue, setInputValue] = useState('');
+
+  // Input settings
+  const [inputStatus, setInputStatus] = useState(''); // Success, error, ''
```

Comment:
Let's use boolean for this.
Initial state will be `null`. When the person starts typying in the input, there is no way back. After this moment there should be only `true` (correct input` or `false` (`incorrect`)

Also let's rename it to `valid`